pallets/public-credentials/src/tests.rs

```diff
@@ -217,76 +168,344 @@ fn add_not_enough_balance() {
-		.with_ctypes(vec![(ctype_hash, attester.clone())])
+		// One less than the minimum required
+		.with_balances(vec![(ACCOUNT_00, deposit - 1)])
 		.build()
 		.execute_with(|| {
 			assert_noop!(
 				PublicCredentials::add(
 					DoubleOrigin(ACCOUNT_00, attester.clone()).into(),
-					Box::new(new_credential.clone())
+					Box::new(new_credential)
 				),
-				Error::<Test>::InvalidClaimerSignature
+				ctype::Error::<Test>::CTypeNotFound
 			);
 		});
 }
 
 #[test]
-fn add_ctype_not_found() {
+fn add_not_enough_balance() {
 	let attester = sr25519_did_from_seed(&ALICE_SEED);
 	let subject_id = SUBJECT_ID_00;
-	let claim_hash = claim_hash_from_seed(CLAIM_HASH_SEED_01);
 	let ctype_hash = get_ctype_hash::<Test>(true);
-	let claimer_id = sr25519_did_from_seed(&CHARLIE_SEED);
 	let new_credential = generate_base_public_credential_creation_op::<Test>(
 		subject_id.into(),
-		claim_hash,
 		ctype_hash,
 		InputClaimsContentOf::<Test>::default(),
-		Some(ClaimerSignatureInfoOf::<Test> {
-			claimer_id: claimer_id.clone(),
-			signature_payload: (claimer_id, hash_to_u8(claim_hash)),
-		}),
 	);
-	let public_credential_deposit: Balance = <Test as Config>::Deposit::get();
-	let attestation_deposit: Balance = <Test as attestation::Config>::Deposit::get();
+	let deposit: Balance = <Test as Config>::Deposit::get();
 
 	ExtBuilder::default()
-		.with_balances(vec![(ACCOUNT_00, public_credential_deposit + attestation_deposit)])
+		// One less than the minimum required
+		.with_balances(vec![(ACCOUNT_00, deposit - 1)])
+		.with_ctypes(vec![(ctype_hash, attester.clone())])
 		.build()
 		.execute_with(|| {
 			assert_noop!(
 				PublicCredentials::add(
 					DoubleOrigin(ACCOUNT_00, attester.clone()).into(),
-					Box::new(new_credential.clone())
+					Box::new(new_credential)
 				),
-				ctype::Error::<Test>::CTypeNotFound
+				Error::<Test>::UnableToPayFees
+			);
+		});
+}
+
+// revoke
+
+#[test]
+fn revoke_successful() {
+	let attester = sr25519_did_from_seed(&ALICE_SEED);
+	let subject_id: <Test as Config>::SubjectId = SUBJECT_ID_00;
+	let new_credential = generate_base_credential_entry::<Test>(ACCOUNT_00, 0, attester.clone(), None);
+	let credential_id: CredentialIdOf<Test> = CredentialIdOf::<Test>::default();
+	let deposit: Balance = <Test as Config>::Deposit::get();
+
+	ExtBuilder::default()
+		.with_balances(vec![(ACCOUNT_00, deposit)])
+		.with_public_credentials(vec![(subject_id, credential_id, new_credential)])
+		.build()
+		.execute_with(|| {
+			assert_ok!(PublicCredentials::revoke(
+				DoubleOrigin(ACCOUNT_00, attester.clone()).into(),
+				credential_id,
+				None,
+			));
+
+			let stored_public_credential_details = Credentials::<Test>::get(&subject_id, &credential_id)
+				.expect("Public credential details should be present on chain.");
+
+			// Test this pallet logic
+			assert!(stored_public_credential_details.revoked);
+
+			// Revoking the same credential does nothing
```

Comment:
Q: Maybe we actually want to throw when double revoking?

Reply:
Yes I thought about that, but I really do not see any issues in revoking something that is already revoked. It won't basically change anything, so I thought it would be fine to simply proceed. What do you think?

Reply:
Yeah its fine. I just wanted to raise awareness such that we are all on the same page. It really does not matter much and makes it easier to implement in the SDK.